pkg/migrations/sql/idempotent/base.sql

```diff
@@ -610,6 +610,23 @@ $$
 LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_PROM.label_key_position(text, text) to prom_reader;
 
+-- public function to delete a metric.
+CREATE OR REPLACE FUNCTION prom_api.delete_metric(metric_name_to_be_deleted text) RETURNS VOID
+AS
+$$
+    DECLARE
+        hypertable_name TEXT;
+        deletable_metric_id INTEGER;
+    BEGIN
+        SELECT table_name, id INTO hypertable_name, deletable_metric_id FROM _prom_catalog.metric WHERE metric_name=metric_name_to_be_deleted;
+        RAISE NOTICE 'deleting "%" metric with metric_id as "%" and table_name as "%"', metric_name_to_be_deleted, deletable_metric_id, hypertable_name;
+        DELETE FROM _prom_catalog.series WHERE metric_id=deletable_metric_id;
```

Comment:
this is better done by dropping the table(partition) SCHEMA_DATA_SERIES.hypertable_name;

Reply:
Ah, I missed that. Updated in the recent commit.